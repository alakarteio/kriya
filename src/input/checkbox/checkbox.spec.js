/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import { mount } from 'enzyme'
import snap from 'snap'
import Checkbox from './checkbox'

const snapshot = snap(Checkbox)

describe('common/Input/Checkbox', () => {
  it('should have a default behaviour', snapshot({}))
  it('should add custom className', snapshot({ className: 'custom' }))
  it('should add custom style', snapshot({ style: { backgroundColor: 'red' } }))
  it('should be checked', snapshot({ input: { checked: true } }))

  describe('callbacks', () => {
    const onClickTest = (checked) => {
      const onClick = jest.fn()
      const wrapper = mount(
        <Checkbox input={{ checked, onChange: onClick }} />,
      )

      wrapper.find('div').first().simulate('click')
      expect(onClick.mock.calls.length).toBe(1)
      expect(onClick.mock.calls[0]).toEqual([!checked])
    }

    it('should swap checked onClick', () => onClickTest(true))
    it('should swap checked onClick', () => onClickTest(false))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
