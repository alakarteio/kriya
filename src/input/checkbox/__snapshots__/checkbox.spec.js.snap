// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Input/Checkbox should add custom className 1`] = `
<div
  className="checkbox custom"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="back"
  />
  <div
    className="ball"
  />
</div>
`;

exports[`common/Input/Checkbox should add custom style 1`] = `
<div
  className="checkbox"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div
    className="back"
  />
  <div
    className="ball"
  />
</div>
`;

exports[`common/Input/Checkbox should be checked 1`] = `
<div
  className="checkbox checked"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="back"
  />
  <div
    className="ball"
  />
</div>
`;

exports[`common/Input/Checkbox should have a default behaviour 1`] = `
<div
  className="checkbox"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="back"
  />
  <div
    className="ball"
  />
</div>
`;
